sdk/src/main/java/com/hcaptcha/sdk/HCaptchaWebViewHelper.java

```diff
@@ -0,0 +1,55 @@
+package com.hcaptcha.sdk;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.graphics.Color;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+
+import androidx.annotation.NonNull;
+
+import com.hcaptcha.sdk.tasks.OnFailureListener;
+import com.hcaptcha.sdk.tasks.OnLoadedListener;
+import com.hcaptcha.sdk.tasks.OnOpenListener;
+import com.hcaptcha.sdk.tasks.OnSuccessListener;
+
+public final class HCaptchaWebViewHelper {
+
+    /**
+     * General setup for the webview:
+     * * enables javascript to be able to load and execute hcaptcha api.js
+     * * loads custom html page to display challenge and/or checkbox
+     */
+    @SuppressLint({"SetJavaScriptEnabled", "AddJavascriptInterface"})
+    public static void prepare(@NonNull HCaptchaConfig config, @NonNull HCaptchaWebViewProvider provider) {
+        WebView webView = provider.getWebView();
+        HCaptchaJSInterface jsInterface = new HCaptchaJSInterface(config, provider, provider, provider, provider);
+        HCaptchaDebugInfo debugInfo = new HCaptchaDebugInfo(webView.getContext());
+        final WebSettings settings = webView.getSettings();
+        settings.setJavaScriptEnabled(true);
+        settings.setLoadWithOverviewMode(true);
+        settings.setCacheMode(WebSettings.LOAD_CACHE_ELSE_NETWORK);
+        webView.setBackgroundColor(Color.TRANSPARENT);
+        webView.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
+        webView.addJavascriptInterface(jsInterface, HCaptchaJSInterface.JS_INTERFACE_TAG);
+        webView.addJavascriptInterface(debugInfo, HCaptchaDebugInfo.JS_INTERFACE_TAG);
+        webView.loadUrl("file:///android_asset/hcaptcha-form.html");
+    }
+
+    public static void finish(@NonNull HCaptchaWebViewProvider provider) {
```

Comment:
`setup` and  `cleanup` might be more appropriate.

Why not use inheritance to use a consistent pattern?
I imagine `HCaptchaHeadlessWebView` and `HCaptchaDialogFragment` can both implement an abstract class `HCaptchaWebViewProvider` which has these `setupWebView` and `cleanupWebView` local methods.
The `HCaptchaWebViewProvider` can also define the local variables common variables between the 2 providers:
```java
@NonNull
private final Handler handler = new Handler(Looper.getMainLooper());

@NonNull
private HCaptchaDialogListener hCaptchaDialogListener;

@NonNull
private final HCaptchaConfig config;

... etc
```

Currently, the 2 providers have some redundancies.

Reply:
> Why not use inheritance for a consistent pattern?

I agree that it will be a perfect solution, but HCaptchaDialogFragment already extends `DialogFragment`.

Probably make sense to convert `"statics" CaptchaWebViewHelper` to object and keep it as a property in both `HCaptchaDialogFragment` and `HCaptchaHeadlessWebView`